rosbag2_performance/rosbag2_performance_benchmarking/src/msg_utils/helpers.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 Apex.AI, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "msg_utils/helpers.hpp"
+
+#include <cstdlib>
+
+namespace msg_utils
+{
+namespace helpers
+{
+void generate_data(rosbag2_performance_benchmarking_msgs::msg::ByteArray & array, size_t size)
+{
+  array.data.reserve(size);
+  for (auto i = 0u; i < size; ++i) {
+    array.data.emplace_back(std::rand() % 255);
+  }
+}
+
+void generate_data(sensor_msgs::msg::Image & msg, size_t size)
+{
+  // TODO(carlossvg): calculate message base size to set total size
```

Comment:
@carlossvg Can you please clarify what this TODO about?

Reply:
@MichaelOrlov Currently the total message size will be all the message fixed size fields (https://github.com/ros2/common_interfaces/blob/rolling/sensor_msgs/msg/Image.msg#L4-L25) + the size of the data field (which is the `size` parameter passed as a function argument). Ideally the total message size would be equal to `size`. In order to do that we would have to calculate the size of all other fields except from `data` and then subtract this value to the `size` parameter to resize `data` the necessary amount.

Reply:
Could you please add meaningful comment to the todo with the same level of details?